xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BootstrapSelect.java

```diff
@@ -67,66 +68,79 @@ public void selectByValue(String value)
 
     public void selectByValues(List<String> values)
     {
-        Map<String, String> valueTitles = new HashMap<>();
-
-        // Get the list of all possible values in the hidden <select> field
-        // WARN: the order of the values is imported since their index will help us to identify
-        // them in the enhanced bootstrap-select widget.
-        List<String> options = new ArrayList<>();
-        for (WebElement element : this.hiddenSelect.findElements(By.tagName("option"))) {
-            String value = element.getAttribute("value");
-            options.add(value);
-            valueTitles.put(value, element.getText());
+        boolean multipleSelect = isMultiple();
+
+        if (!multipleSelect && values.size() > 1) {
+            throw new AssertionError("More than one value specified for a select that does not allow multiple values.");
         }
 
-        // Open the enhanced bootstrap-select widget
-        WebElement menu = openMenu();
+        List<String> valuesToToggle = new ArrayList<>();
 
-        if (isMultiple()) {
-            // For each of its option
-            for (WebElement element : menu.findElements(By.tagName("li"))) {
-                // We find the language associated to this item thanks to the attribute "data-original-index"
-                int index = Integer.parseInt(element.getAttribute("data-original-index"));
-                String value = options.get(index);
+        // The list of values is usually small while the list of all options can be huge - use one XPath query per
+        // value.
+        for (String value : values) {
+            assertFalse(value.contains("\""), "Value must not contain a double quote");
```

Comment:
Feels a bit weird in PO but I understand why with the following line.

Reply:
Yeah, and there is unfortunately no way to escape quotes in XPath, see [Apostrophe (') in XPath query - StackOverflow](https://stackoverflow.com/questions/1341847/apostrophe-in-xpath-query).